final_task/rss_reader/converting.py

```diff
@@ -0,0 +1,146 @@
+import logging as log
+import requests as r
+import check
+import os
+import datetime
+import sys
+
+from fpdf import FPDF
+from dominate import document
+from dominate.tags import div, h2, img, p
+
+import printers
+
+now = datetime.datetime.now
+
+
+def create_html(items: list) -> document:
+    """
+        convert article data in html format
+    """
+    html_document = document(title='Dominate your HTML')
+    log.info('Start make html format')
+    for item in items:
+        item = printers.prepare_one_item(item)
+        with html_document:
+            with div():
+                h2("Title: " + item['Title:'])
+                p("Link: " + item['Link: '])
+                img(src=item['Media content:\n'])
+                p("Description: " + item['Description: '])
+                p("Date: " + item['Date:'])
+    return html_document
+
+
+def write_to_file(items: list) -> None:
+    result = create_html(items)
+    data = now().strftime("%d-%m-%Y")
+    time = now().strftime("%X")
+    name_of_html = str(data) + '_' + time
+    with open("{}".format(name_of_html), "w") as f:
+        f.write(str(result))
+        log.info("Successful converting into html")
+
+
+def create_pdf(items: list) -> None:
+    log.info("Start creating pdf")
+    date = now().strftime("%d-%m-%Y")
+    time = now().strftime("%X")
+    pdf = FPDF()
+    element = str
+    img_path = str
+    try:
+        pdf.add_page()
+        pdf.add_font("DejaVu", "", "DejaVuSans.ttf", uni=True)
+        pdf.set_font("DejaVu", "", 10)
+    except RuntimeError:
+        log.info("There isn't DejaVuSans.ttf in your rep")
+        print("Something go wrong, check DejaVuSans.ttf in your rep ")
+        sys.exit()
+    pdf.write(8, "RSS feed")
+    pdf.ln(20)
+
+    for item in items:
+        item = printers.prepare_one_item(item)
+        pdf.write(8, "===Wow! News!===")
+        pdf.ln(10)
+        pdf.write(8, "Title: " + str(item['Title:']))
+        pdf.ln(10)
+        pdf.write(8, "Link: " + str(item['Link: ']))
+        pdf.ln(15)
+        pdf.write(8, "Date: " + str(item["Date:"]))
+        pdf.ln(10)
+        img_url = str(item['Media content:\n'])
+        url_list = img_url.split("\n")
+
+        # There's a difficult moment, cause 'fpdf' lib can't
+        # get images from the Internet
+        # So we should use temporary files
+
+        if check.internet_on() and img_url != '':
+            log.info("There is connection")
+            try:
+                for element in url_list:
+                    # Try to understand format of image file
+                    log.info("Try to understand format of image file")
+                    if element.endswith(".png") or element.endswith(".jpg"):
+                        image = r.get(element)
+                        try:
+                            img_path = str(len(element)) + \
+                                       str(img_url[-4:])
```

Comment:
It is better to use `os.path` for operations with filesystem paths because Windows OS and Linux OS have different separators